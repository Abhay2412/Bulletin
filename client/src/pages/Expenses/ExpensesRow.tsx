import React from "react";
import { useState } from "react";
import { Box, Button, Select, MenuItem } from "@mui/material";

interface Props {
  name: string;
  date: string;
  assigned: string;
  balance: number;
}

const mockMemberInfo = [
  { id: 1, name: "Liane Doe", email: "liane.doe@gmail.com", role: "Admin" },
  { id: 2, name: "Dad Doe", email: "dad.doe@gmail.com", role: "Admin" },
  { id: 3, name: "Logan Doe", email: "logan.doe@gmail.com", role: "Member" },
  { id: 4, name: "Aly Doe", email: "aly.doe@gmail.com", role: "Member" },
];

export default function ExpensesRow({ name, date, assigned, balance }: Props) {

  return (
    <>
      <Box className="expensesTableRow tableCellsFormatting" width="20%">
        {name}
      </Box>
      <Box className="expensesTableRow tableCellsFormatting" width="15%">
        {date}
      </Box>

      <Box className="expensesTableRow tableCellsFormatting" width="35%">
        {assigned}
      </Box>
      <Box className="expensesTableRow tableCellsFormatting" width="20%">
        ${balance.toFixed(2)}
      </Box>
      <Box className="expensesTableRow" width="5%">
        <Button>X</Button>
      </Box>
      <Box className="expensesTableRow" width="5%">
        <Button sx={{ textDecoration: "underline" }}>edit</Button>
      </Box>
    </>
  );
}
