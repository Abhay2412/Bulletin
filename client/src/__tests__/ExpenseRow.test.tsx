import React from "react";
import ExpensesRow from "../pages/Expenses/ExpensesRow";
import { configure, mount, ReactWrapper } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import renderer from "react-test-renderer";
import { store } from "../store";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { format, parse } from "date-fns";

configure({ adapter: new Adapter() });

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useParams: () => ({
    board_id: "g7HGuvfnGe3nhQgAF8qY",
  }),
}));

describe("Testing <ExpenseRow /> Component", () => {
  const wrapper: ReactWrapper = mount(
    <Provider store={store}>
      <BrowserRouter>
        <ExpensesRow
          name="Test Expense"
          date={parseInt(
            format(parse("2022-03-28", "yyyy-MM-dd", new Date()), "T")
          )}
          assigned="Test User"
          balance={200}
          type="Expense"
          boardId="3289uweihdkaj"
          expenseId="wpqoedwjfsax"
          expenses={[]}
          setExpenses={() => {}}
          budgets={[]}
          setBudgets={() => {}}
        />
      </BrowserRouter>
    </Provider>
  );
  it("should render ExpenseRow properly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <ExpensesRow
              name="Test Expense"
              date={parseInt(
                format(parse("2022-03-28", "yyyy-MM-dd", new Date()), "T")
              )}
              assigned="Test User"
              balance={200}
              type="Expense"
              boardId="3289uweihdkaj"
              expenseId="wpqoedwjfsax"
              expenses={[]}
              setExpenses={() => {}}
              budgets={[]}
              setBudgets={() => {}}
            />
          </BrowserRouter>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render BudgetRow properly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <ExpensesRow
              name="Test Expense"
              date={parseInt(
                format(parse("2022-03-28", "yyyy-MM-dd", new Date()), "T")
              )}
              assigned="Test User"
              balance={200}
              type="Expense"
              boardId="3289uweihdkaj"
              expenseId="wpqoedwjfsax"
              expenses={[]}
              setExpenses={() => {}}
              budgets={[]}
              setBudgets={() => {}}
            />
          </BrowserRouter>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should open an editable modal", () => {
    wrapper
      .find("button")
      .find("#edit-expense-dialog-button")
      .simulate("click");
    expect(wrapper.find(".overlayBox").exists()).toBeTruthy();
  });

  describe("should have edit modal open", () => {
    beforeAll(() => {
      wrapper
        .find("button")
        .find("#edit-expense-dialog-button")
        .simulate("click");
    });

    it("should change expense name field value", () => {
      wrapper
        .find("input")
        .find("#expenses-overlay-name-field")
        .simulate("change", {
          target: { value: "Test Expense" },
        });
      expect(
        wrapper.find("input").find("#expenses-overlay-name-field").get(0).props
          .value
      ).toEqual("Test Expense");
    });
  });

  it("should close modal", () => {
    wrapper
      .find("button")
      .find("#expense-modal-close-button")
      .simulate("click");
    expect(wrapper.find(".overlayBox").exists()).toBeFalsy();
  });
});
