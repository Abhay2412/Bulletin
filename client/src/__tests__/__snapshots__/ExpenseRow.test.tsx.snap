// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <ExpenseRow /> Component should render BudgetRow properly 1`] = `
Array [
  <div
    className="expensesTableRow tableCellsFormatting MuiBox-root css-ufscxh"
  >
    Test Expense
  </div>,
  <div
    className="expensesTableRow tableCellsFormatting MuiBox-root css-w56oqh"
  >
    2022-03-28
  </div>,
  <div
    className="expensesTableRow tableCellsFormatting MuiBox-root css-2rhtyh"
  >
    Test User
  </div>,
  <div
    className="expensesTableRow tableCellsFormatting MuiBox-root css-ufscxh"
  >
    $
    200.00
  </div>,
  <div
    className="expensesTableRow MuiBox-root css-ndbrsi"
  >
    <button
      className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      X
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>,
  <div
    className="expensesTableRow MuiBox-root css-ndbrsi"
  >
    <button
      className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-taz78y-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      id="edit-expense-dialog-button"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      edit
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>,
]
`;

exports[`Testing <ExpenseRow /> Component should render ExpenseRow properly 1`] = `
Array [
  <div
    className="expensesTableRow tableCellsFormatting MuiBox-root css-ufscxh"
  >
    Test Expense
  </div>,
  <div
    className="expensesTableRow tableCellsFormatting MuiBox-root css-w56oqh"
  >
    2022-03-28
  </div>,
  <div
    className="expensesTableRow tableCellsFormatting MuiBox-root css-2rhtyh"
  >
    Test User
  </div>,
  <div
    className="expensesTableRow tableCellsFormatting MuiBox-root css-ufscxh"
  >
    $
    200.00
  </div>,
  <div
    className="expensesTableRow MuiBox-root css-ndbrsi"
  >
    <button
      className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      X
    </button>
  </div>,
  <div
    className="expensesTableRow MuiBox-root css-ndbrsi"
  >
    <button
      className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-taz78y-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      id="edit-expense-dialog-button"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      edit
    </button>
  </div>,
]
`;
